docs/source/en/tasks/object_detection.mdx

```diff
@@ -0,0 +1,487 @@
+Object detection models are commonly evaluated with a set of [COCO-style metrics](https://cocodataset.org/#detection-eval).
+You can use one of the existing metrics implementations, here you'll use the one from `torchvision`.
+
+Save your trained model to use for evaluation. Alternatively, you can evaluate the model that you have pushed to the Hub.
+
+```py
+>>> trainer.save_model("./detr-resnet-50_fine_tuned_cppe5")
+```
+
+To use the `torchvision` evaluator, you'll need to prepare a ground truth COCO dataset. The API to build a COCO dataset
+requires the data to be stored in a certain format, so you'll need to save images and annotations to disk first. Just like
+when you prepared your data for training, the annotations from the `cppe5["test"]` need to be formatted. However, images
+should stay as they are.
+
+The following example shows how you can:
+- Create a torchvision CocoDetection dataset
+- Reformat annotations from the `cppe5["test"]`
+- Save images and annotations into files that you can create an instance of CocoDetection dataset from
+- Use `cocoevaluate` metric to compute the [COCO metrics](https://cocodataset.org/#detection-eval).
+
+```py
+import os
+import json
+import evaluate
+import torchvision
+import numpy as np
+from tqdm import tqdm
+
+
+class CocoDetection(torchvision.datasets.CocoDetection):
+    def __init__(self, img_folder, feature_extractor, ann_file):
+        super(CocoDetection, self).__init__(img_folder, ann_file)
+        self.feature_extractor = feature_extractor
+
+    def __getitem__(self, idx):
+        # read in PIL image and target in COCO format
+        img, target = super(CocoDetection, self).__getitem__(idx)
+
+        # preprocess image and target (converting target to DETR format, resizing + normalization of both image and target)
+        image_id = self.ids[idx]
+        target = {"image_id": image_id, "annotations": target}
+        encoding = self.feature_extractor(images=img, annotations=target, return_tensors="pt")
+        pixel_values = encoding["pixel_values"].squeeze()  # remove batch dimension
+        target = encoding["labels"][0]  # remove batch dimension
+
+        return {"pixel_values": pixel_values, "labels": target}
+
+
+im_processor = DetrImageProcessor.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
+model = DetrForObjectDetection.from_pretrained("./detr-resnet-50_fine_tuned_cppe5")
+
+# prepare the test dataset
+def val_formatted_anns(image_id, objects):
+    annotations = []
+    for i in range(0, len(objects["id"])):
+        new_ann = {
+            "id": objects["id"][i],
+            "category_id": objects["category"][i],
+            "iscrowd": 0,
+            "image_id": image_id,
+            "area": objects["area"][i],
+            "bbox": objects["bbox"][i],
+        }
+        annotations.append(new_ann)
+
+    return annotations
+
+
+# CocoDetection expects to read the data from files in a certain format
+def save_cppe5_annotation_file_images(cppe5):
```

Comment:
Since this snippet is long, consider grouping semantically close lines of code together and putting comments on top of them.

Reply:
Fixed in 1a38260b27ef977e0d82ba1c918eeb19c8f3dcb7